pkg/cache/server/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kcp-dev/logicalcluster/v2"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+
+	configcrds "github.com/kcp-dev/kcp/config/crds"
+)
+
+var SystemCRDLogicalCluster = logicalcluster.New("system:system-crds")
+
+func Bootstrap(ctx context.Context, logger logr.Logger, apiExtensionsClusterClient apiextensionsclient.ClusterInterface) error {
+	crds := []*apiextensionsv1.CustomResourceDefinition{}
+	for _, resource := range []string{"apiresourceschemas", "apiexports"} {
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		if err := configcrds.Unmarshal(fmt.Sprintf("apis.kcp.dev_%s.yaml", resource), &crd); err != nil {
+			panic(fmt.Errorf("failed to unmarshal %v resource: %v", resource, err))
+		}
+		for i := range crd.Spec.Versions {
+			v := &crd.Spec.Versions[i]
+			v.Schema = &apiextensionsv1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Type:                   "object",
+					XPreserveUnknownFields: pointer.BoolPtr(true),
+				},
+			} // wipe the schema, we don't need validation
+		}
+		crds = append(crds, crd)
+	}
+
+	return wait.PollInfiniteWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+		for _, crd := range crds {
+			_, err := apiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster).ApiextensionsV1().CustomResourceDefinitions().Create(ctx, crd, metav1.CreateOptions{})
```

Comment:
creating crds with empty schemas is cheap and doesn't require creating an `in-memory` storage implementation.

it can be reworked in the future if we don't like it.

Reply:
the storage is required for listing `CRDs` and for discovery (not implemented yet).